plugins/TwoFactorAuth/TwoFactorAuth.php

```diff
@@ -180,14 +180,11 @@ private function requiresAuth($module, $action, $parameters)
             return false;
         }
 
-        if (Piwik::getModule() === 'Widgetize') {
-            // we cannot use $module as it would be different when dispatching other requests within the widgetized request
-            $auth = StaticContainer::get('Piwik\Auth');
-            if ($auth && !$auth->getLogin() && method_exists($auth, 'getTokenAuth') && $auth->getTokenAuth()) {
-                // when authenticated by token only, we do not require 2fa
-                // needed eg for rendering exported widgets authenticated by token
-                return false;
-            }
+        $auth = StaticContainer::get('Piwik\Auth');
+        if ($auth && !$auth->getLogin() && method_exists($auth, 'getTokenAuth') && $auth->getTokenAuth()) {
+            // when authenticated by token only, we do not require 2fa
+            // needed eg for rendering exported widgets authenticated by token
+            return false;
```

Comment:
Wouldn't that actually allow to bypass the 2fa everywhere when using the tokenauth?

Reply:
That should be fine though. You basically can't do 2FA for the token_auth and that's already the behaviour for the API as well. We just need to make sure the user is actually authenticated using the token and not through regular login/session.

Not sure but could also add another check in the `if` like `Piwik::isUserHasAtLeastSomeViewAccess()` but not sure if needed or if it makes sense.